### PR TITLE
Submit clinical additions

### PR DESCRIPTION
**Pull request by ciaranschutte**
- add signoff modal
- add gql error logging
- some table config and styling cleanup

### CODE REVIEW

File: src/app/(post-login)/submission/program/[shortName]/clinical-submission/page.tsx [hunk 1]
```diff
@@ -338,6 +388,18 @@ const ClinicalSubmissionPage = ({
 
     return (
       <>
+        {signOffModalShown && (
+          <ModalPortal>
+            <SignOffValidationModal
+              //hasUpdate={hasUpdate || true}
```
Comment (demariadaniel) [inline src/app/(post-login)/submission/program/[shortName]/clinical-submission/page.tsx:394]: This can be removed

**Comment by ciaranschutte** [comment]
> Tons of pipeline (linting?) issues, maybe the repo's not there yet but are there any related to this specific feature we can clean up?

Taking a look now. It's building ok locally and I didn't see any pipeline errors.

There will likely be linting warnings that can be improved upon later.
It's already a lot of work applying the current TS standards to this old code, if we are to cover every linting rule also that's a lot low priority work to tackle.

**Comment by ciaranschutte** [comment]
> Tons of pipeline (linting?) issues, maybe the repo's not there yet but are there any related to this specific feature we can clean up?

yes good idea. i'll take a look.

**Comment by demariadaniel** [comment]
> There will likely be linting warnings that can be improved upon later. It's already a lot of work applying the current TS standards to this old code, if we are to cover every linting rule also that's a lot low priority work to tackle.

OK, just anything that's related to this feature and reduces later cleanup before it piles up too much.

**Review by demariadaniel — APPROVED** [review]

**Review by demariadaniel — APPROVED** [review]

File: src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/FilesNavigator/StatsArea.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { ComponentProps } from "react";
 
 const StarIcon = DataTableStarIcon;
 
-type RecordState = "NEW" | "NONE" | "UPDATED" | "ERROR" | "WARNING";
+export type RecordState = "NEW" | "NONE" | "UPDATED" | "ERROR" | "WARNING";
```
Comment (joneubank) [inline src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/FilesNavigator/StatsArea.tsx:11]: What's the logic to this ordering? not alphabetical or logical as far as I can tell. Maybe sort it alphabetically instead?

File: src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/FilesNavigator/TableCells.tsx [hunk 3]
```diff
@@ -82,42 +75,38 @@ export const DataFieldCell = ({
   file,
 }: {
   original: FileRecord;
-  fieldName: string;
+  fieldName: keyof FileRecord;
```
Comment (joneubank) [inline src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/FilesNavigator/TableCells.tsx:78]: 👍

File: src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/SubmissionSummaryTable.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+type SubmissionSummaryStatus = {
+  submissionSummaryStatus: string;
+};
+
+// This type allows us to have a subset of the GqlClinicalEntity properties as columns, but not any other columns.
+//   A SubmissionSummaryStatus column is also allowed.
+type SubmissionSummaryColumns = Partial<
+  SubmissionSummaryStatus & {
+    [k in ClinicalEntityType]: string;
```
Comment (joneubank) [inline src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/SubmissionSummaryTable.tsx:41]: Optionally you could write this as `Partial<SubmissionSummaryStatus & Record<ClinicalEntityType, string>>`

This is a personal preference thing but I find `Record<KeyType, ValueType>` easier to read than the `{ [k in KeyValues]: ValueType }` pattern.

File: src/app/(post-login)/submission/program/[shortName]/clinical-submission/page.tsx [hunk 5]
```diff
@@ -303,7 +308,46 @@ const ClinicalSubmissionPage = ({
...
     };
-    const handleSignOff = () => new Promise(() => true);
+
+    const handleSignOff = async () => {
+      try {
+        const userDidApprove = await getSignOffConfirmation();
+
+        if (userDidApprove) {
+          setGlobalLoading(true);
+          await sleep();
```
Comment (joneubank) [inline src/app/(post-login)/submission/program/[shortName]/clinical-submission/page.tsx:318]: feels like a comment explaining why we're sleeping would be helpful here.

**Pull request merged**